http.c

```diff
@@ -2775,6 +2781,246 @@ evhttp_connection_set_closecb(struct evhttp_connection *evcon,
+	} else
+		return -1;
+	if (send(socks5->fd, msg, lenght, 0) <= 0)
+		return -2;
+
+	return 0;
+}
+
+static int
+set_socks5_address(struct context_socks5 *socks5, const char *address, unsigned short port)
+{
+	socks5->port = port;
+	if (!evutil_inet_pton(AF_INET, address, &socks5->addr4.sin_addr))
+		strcpy(socks5->domain_name, address);
+	else
+		strcpy(socks5->ipv4, address);
+	return 0;
+}
+
+static int
+connect_socks5(struct context_socks5 *socks5)
+{
+	unsigned char tmp[512];
+	if (3 != send(socks5->fd, "\x05\x01\x00", 3, 0))
+		return -1;
+	if (recv(socks5->fd, tmp, 512, 0) != 2)
+		return -2;
+	if (read_socks5_reponse(tmp[1]))
+		return -3;
+	return 0;
+}
+
+static int
+connnect_target_over_socks5(struct context_socks5 *socks5)
+{
+	int ret;
+	unsigned char tmp[512];
+	if (request_socks5_target(socks5))
+		return -1;
+	ret = recv(socks5->fd, tmp, 512, 0);
+	if (ret < 0 || tmp[1] != 0)
+		return -2;
+	return 0;
+}
+
+// just support no auth socks5 proxy. Such as 'ssh -D'.
+static int
+socks5_handshake(int fd, const char *szTargetHost, unsigned short port)
+{
+	int bOK = -1;
+	struct context_socks5 s5;
+	memset(&s5, 0, sizeof(struct context_socks5));
+	do {
+		// init
+		s5.fd = fd;
+		// do
+		if (set_socks5_address(&s5, szTargetHost, port)) 
+			break;
+		if (connect_socks5(&s5)) 
+			break;
+		if (connnect_target_over_socks5(&s5)) 
+			break;
+		bOK = 0;
+	} while (0);
+	return bOK;
+}
+
+static int
+evhttp_proxy_request_httpconnect(int fd, struct evhttp_connection *evcon)
+{
+	char buf_conn[512];
+	int ret = -1, ret_child = 0;
+	do {
+		snprintf(buf_conn,sizeof(buf_conn),
+			"CONNECT %s:%d HTTP/1.0\r\nHost: "
+			"%s:%d\r\nProxy-Connection: Keep-Alive\r\nPragma: no-cache\r\n\r\n",
+			evcon->proxy_target_address, evcon->proxy_target_port, evcon->proxy_address, evcon->proxy_port);
+		if ((ret_child = send(fd, buf_conn, strlen(buf_conn), 0)) <= 0)
+			break;
+		memset(buf_conn, 0, sizeof(buf_conn));
+		if ((ret_child = recv(fd, buf_conn, sizeof(buf_conn), 0)) < 32)
+			break;
+		if (!strstr(buf_conn, "HTTP/1.1 200 ") && !strstr(buf_conn, "HTTP/1.0 200 "))
+			break;
+		ret = 0;
+	} while (0);
+	return ret;
+}
+
+static int
+evhttp_proxy_connect_callback_internal(struct evhttp_connection *evcon, void *arg)
+{
+	struct bufferevent *bev = evhttp_connection_get_bufferevent(evcon);
+	int nRet = -1, fd = bufferevent_getfd(bev), orig_flag,
+		flags = fcntl(fd, F_GETFL, 0);
+	orig_flag = flags;
+	do {
+		// write timeout check
+		fd_set w;
+		struct timeval timeout = {};
+		timeout.tv_sec = 3;
+		FD_ZERO(&w);
+		FD_SET(fd, &w);
+		if (select(fd + 1, NULL, &w, NULL, &timeout) <= 0)
+			break;
+		FD_CLR(fd, &w);
+		// must
+		fcntl(fd, F_SETFL, flags & ~O_NONBLOCK); // set block
+		// choice
+		if (1 == evcon->proxy_type){
+			if (evhttp_proxy_request_httpconnect(fd,evcon))
+				break;
+		}else if (2 == evcon->proxy_type){
+			if (socks5_handshake(fd, evcon->proxy_target_address,evcon->proxy_target_port))
+				break;
+		}else
```

Comment:
There are lots of style issues like this in the code

```suggestion
		} else
```